overlord/hookstate/hookmgr.go

```diff
@@ -249,6 +257,18 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		}
 	}
 
+	if hookExists || mustHijack {
+		// we will run something, not a noop
+		if task.State().Restarting() {
+			// don't start running a hook if we are restarting
+			return &state.Retry{}
+		}
+
+		// keep count of running hooks
+		atomic.AddInt32(&m.runningHooks, 1)
```

Comment:
If hook is hijacked do we actually run any hooks? I think it's just an internal function then?

Reply:
well, the question is also whether that internal code would be killed by the tomb or not (it's not at the moment but that could change), it seems easier to be consistent this way